### ISSUE TITLE
Unused value assignment analysis doesn't handle ref locals

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#32855]
**Version Used**: Visual Studio 2019 Preview 2

Found while working on https://github.com/tonerdo/coverlet/pull/309

**Steps to Reproduce**:

```csharp
class Test
{
  int[] data = { 0 };

  void Method()
  {
    ref int target = ref data[0];
    target = 1;
  }
}
```

**Expected Behavior**:

IDE0059 is not reported for the assignment.

**Actual Behavior**:

IDE0059 is reported for the assignment.

**Comment by Tragetaschen** [comment]
Does this fix this as well?
```csharp
private void test()
{
    ReadOnlySequence<byte> a = default;
    var count = 0;
    foreach (var _ in a) // IDE0059 smudge here on _
        ++count;
}
```
